resnet.c

```diff
@@ -258,12 +258,22 @@ int getnetinfo(int res_id, void *out, void *hint, int pid, int flags)
 	int ver = 0;
 	char *p;
 
+#define CHECK_SIZE(sz, req_sz)						\
+	if (sz < req_sz) {						\
+		eprintf("memory (%ld) is not enough to hold data (%ld)",\
+			sz, req_sz);					\
+		errno = ENOMEM;						\
+		return -1;						\
+	}
+
 	switch (res_id) {
 	case RES_NET_IFSTAT:
+		CHECK_SIZE(sz, sizeof(res_net_ifstat_t));
+
 		/* Interface name should be provided */
 		if (hint == NULL) {
 			eprintf(
-				"Interface name is not provided for RES_NET_IFSTAT");
```

Comment:
Just curious, why did you drop this?

Reply:
I was thinking I don't need to provide resource ID name as application will have it while calling.
But keeping it makes it easier for application so putting this back with line length fix.